### PR TITLE
Add Headers and body to API Transports

### PR DESCRIPTION
**Pull request by mendoza-conicet**
add PUT Method

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/Alert/Transport/Api.php [hunk 1]
```diff
@@ -63,10 +73,20 @@ private function contactAPI($obj, $api, $options, $method, $auth)
         $client = new \GuzzleHttp\Client();
         if (isset($auth) && !empty($auth[0])) {
             $request_opts['auth'] = $auth;
-        }
+	}
+	if (is_array($request_heads)) {
```
Comment (Jellyfrog) [inline LibreNMS/Alert/Transport/Api.php]: Will always be true

File: LibreNMS/Alert/Transport/Api.php [hunk 2]
```diff
@@ -63,10 +73,20 @@ private function contactAPI($obj, $api, $options, $method, $auth)
...
+		$request_opts['headers'] = $request_heads;
+	}
+	if (strlen($body))
+		$request_opts['body'] = $body;
+
+        $request_opts['query'] = $query;
```
Comment (Jellyfrog) [inline LibreNMS/Alert/Transport/Api.php]: Why this here?

File: LibreNMS/Alert/Transport/Api.php (hunk 1 above)
Comment (mendoza-conicet) [reply LibreNMS/Alert/Transport/Api.php]: fixed

File: LibreNMS/Alert/Transport/Api.php (hunk 2 above)
Comment (mendoza-conicet) [reply LibreNMS/Alert/Transport/Api.php]: a bad commit, sorry

**Comment by PipoCanaja** [comment]
Hi @mendoza-conicet  

Thanx for your PR ! 

You should probably also add the variable replacement for the $body variable. You should also adapt the documentation and explain that "body" makes sense only for PUT. 

And you also need to sign the Contributor Licence Agreement it seems (That is mandatory for us to merge the PR)

PipoCanaja

**Comment by PipoCanaja** [comment]
Hi @mendoza-conicet 
Could you please update the documentation as well to explain how to use these new options, and add examples as it is done for existing ?
Thanx

**Comment by laf** [comment]
You've also got a code issue which Travis is showing: https://travis-ci.com/librenms/librenms/jobs/247960201#L441

**Comment by laf** [comment]
@mendoza-conicet Just docs now :)

**Comment by PipoCanaja** [comment]
Hi @mendoza-conicet 
Thanx for your contribution. Your PR will be merged tomorrow (tests are running right now). 
Bye

**Comment by PipoCanaja** [comment]
LGTM, Thanx

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-58-release-changelog-november-2019/10224/1

**Pull request merged**